apps/src/dance/ai/DanceAiModalHeader.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {useMemo} from 'react';
+import moduleStyles from './dance-ai-modal.module.scss';
+import aiBotMini from '@cdo/static/dance/ai/bot/ai-bot-mini.svg';
+const i18n = require('../locale');
+
+import {DanceAiModelItem} from './types';
+
+import EmojiIcon from './EmojiIcon';
+
+interface DanceAiModalHeaderProps {
+  onClose: () => void;
+  handleStartOverClick: () => void;
+  selectedEmojis: DanceAiModelItem[];
+  slotCount: number;
+}
+
+const DanceAiModalHeader: React.FunctionComponent<DanceAiModalHeaderProps> = ({
+  onClose,
+  handleStartOverClick,
+  selectedEmojis,
+  slotCount,
+}) => {
+  const headerValue = useMemo(() => {
+    const filledSelectedEmojis: (DanceAiModelItem | undefined)[] = [
+      ...selectedEmojis,
+    ];
+    filledSelectedEmojis.length = slotCount;
+    filledSelectedEmojis.fill(undefined, selectedEmojis.length, slotCount);
```

Comment:
our `selectedEmojis` is explicitly filtered down to not contain any undefined values. Which is fine in most places, except here we always want to display the empty slots.

So we copy our selectedEmojis to a new array, set the length to make it bigger, and fill in everything after the selected emoji length to the slotcount with undefined. We have to fill because we can't map over empty items.

Then down below, we iterate over the filled list and display them. The only gotcha is that we use the `item.id` as our key where we can, but if it doesn't exist we fall back on the index.

Reply:
I wonder if it would be better to just rework this with CSS, instead of requiring that we always have three elements?